shunting-yard-common/src/test/java/com/hotels/shunting/yard/common/messaging/MessageTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.common.messaging;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import org.junit.Test;
+
+public class MessageTest {
+
+  @Test
+  public void checkNull() {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload(null);
+    objectUnderTest.timestamp(1515585600000L);
+
+    assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> objectUnderTest.build())
+        .withMessage("Parameter 'payload' is required");
+  }
+
+  @Test
+  public void checkEmpty() {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+
+    assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> objectUnderTest.build())
+        .withMessage("Parameter 'table' is required");
+  }
+
+  @Test
+  public void testBuild() {
```

Comment:
I don't think this test adds anything beyond testGetters. I'd rename testGetters to testBuild and add the isNotNull assertion there.

Reply:
@peterschrammel done